src/privatesend/privatesend.h

```diff
@@ -53,27 +53,64 @@ enum PoolMessage {
     MSG_SUCCESS,
     MSG_ENTRIES_ADDED,
     ERR_SIZE_MISMATCH,
-    MSG_POOL_MIN = ERR_ALREADY_HAVE,
-    MSG_POOL_MAX = ERR_SIZE_MISMATCH
 };
+template<> struct is_serializable_enum<PoolMessage> : std::true_type {};
 
 // pool states
-enum PoolState {
+enum PoolState : int32_t {
     POOL_STATE_IDLE,
     POOL_STATE_QUEUE,
     POOL_STATE_ACCEPTING_ENTRIES,
     POOL_STATE_SIGNING,
     POOL_STATE_ERROR,
-    POOL_STATE_SUCCESS,
-    POOL_STATE_MIN = POOL_STATE_IDLE,
-    POOL_STATE_MAX = POOL_STATE_SUCCESS
+    POOL_STATE_SUCCESS
 };
+template<> struct is_serializable_enum<PoolState> : std::true_type {};
 
 // status update message constants
-enum PoolStatusUpdate {
+enum PoolStatusUpdate : int32_t {
     STATUS_REJECTED,
     STATUS_ACCEPTED
 };
+template<> struct is_serializable_enum<PoolStatusUpdate> : std::true_type {};
+
+class CPrivateSendStatusUpdate
+{
+public:
+    int nSessionID;
+    PoolState nState;
+    int nEntriesCount; // deprecated, kept for backwards compatibility
+    PoolStatusUpdate nStatusUpdate;
+    PoolMessage nMessageID;
+
+    CPrivateSendStatusUpdate() :
+        nSessionID(0),
+        nState(POOL_STATE_IDLE),
```

Comment:
BTW, I'm a big fan of [default initializers for class members](https://arne-mertz.de/2015/08/new-c-features-default-initializers-for-member-variables/), which IMHO gives much easier to read code and provides much more secure code (introducing a new constructor does not by accident leave out member initialization).

We don't have this as part of our code style guidelines, but maybe we should introduce it (assuming you also like the new style of initialization). Obviously, we won't easily adopt that style to old Bitcoin code...but doing it in our own code is at least something...